### ISSUE TITLE
`cast wallet address --keystore` throwing error

### ISSUE THREAD

**Issue by 0xbok** [foundry-rs/foundry#4629]
### Component

Cast

### Have you ensured that all of these are up to date?

- [X] Foundry
- [X] Foundryup

### What version of Foundry are you on?

forge 0.2.0 (6570a4c 2023-03-23T00:12:05.600914Z)

### What command(s) is the bug in?

cast wallet address --keystore

### Operating System

macOS (Apple Silicon)

### Describe the bug

`cast wallet address --keystore <keystore>` is throwing this error:
```
Error:
Error accessing local wallet. Did you set a private key, mnemonic or keystore?
Run `cast send --help` or `forge create --help` and use the corresponding CLI
flag to set your key via:
--private-key, --mnemonic-path, --aws, --interactive, --trezor or --ledger.

Alternatively, if you're using a local node with unlocked accounts,
use the --unlocked flag and set the `ETH_FROM` environment variable to the address
of the unlocked account you want to use.
```

It used to ask for password earlier after which it displayed the address.

**Comment by nzhl** [comment]
Similar problem encoutered. 

```
# ./mm.text
invite remember gate measure double jazz range logic equip goddess artwork someone

> cast wallet sign --mnemonic ./mm.text "hello"    

> Error:
Error accessing local wallet. Did you set a private key, mnemonic or keystore?
Run `cast send --help` or `forge create --help` and use the corresponding CLI
flag to set your key via:
--private-key, --mnemonic-path, --aws, --interactive, --trezor or --ledger.

Alternatively, if you're using a local node with unlocked accounts,
use the --unlocked flag and set the `ETH_FROM` environment variable to the address
of the unlocked account you want to use.

>  cast wallet sign --mnemonic "invite remember gate measure double jazz range logic equip goddess artwork someone
" "hello"   

> Error:
Error accessing local wallet. Did you set a private key, mnemonic or keystore?
Run `cast send --help` or `forge create --help` and use the corresponding CLI
flag to set your key via:
--private-key, --mnemonic-path, --aws, --interactive, --trezor or --ledger.

Alternatively, if you're using a local node with unlocked accounts,
use the --unlocked flag and set the `ETH_FROM` environment variable to the address
of the unlocked account you want to use. 

```

**Comment by mattsse** [comment]
thanks for flagging, this was indeed introduced recently, sorry about that,
should be fixed by #4641
